utils/io/fileutils/archive.go

```diff
@@ -166,18 +168,25 @@ func checkSymlinkEntry(header *archiveHeader, archiveEntry archiver.File, destin
 	}
 
 	if !isEntryInDestination(destinationDir, targetLinkPath) {
-		return errorutils.CheckError(errors.New("illegal link path in archive: " + targetLinkPath))
+		return errorutils.CheckError(fmt.Errorf(
+			"illegal link path in archive: '%s'. For security reasons, the path should lead to an entry under '%s'",
+			targetLinkPath, destinationDir))
 	}
 	return nil
 }
 
 // Make sure the extraction path of the archive entry is under the destination dir
 func isEntryInDestination(destinationDir, pathInArchive string) bool {
+	// If pathInArchive starts with '/' and we are on Windows, the path is illegal
+	pathInArchive = strings.TrimSpace(pathInArchive)
+	if os.IsPathSeparator('\\') && strings.HasPrefix(pathInArchive, "/") {
+		return false
+	}
 	// Since the entry in archive should be always represented as Unix path, the "path" module is used and not "filepath"
```

Comment:
Looks like this comment is no longer valid, following the replacement of ```path``` with ```filepath```. Is this correct?